src/main/java/fr/zcraft/zlib/components/gui/ArrayPromptGui.java

```diff
@@ -0,0 +1,93 @@
+package fr.zcraft.zlib.components.gui;
+
+import org.bukkit.entity.Player;
+import org.bukkit.inventory.ItemStack;
+
+import fr.zcraft.zlib.components.gui.ExplorerGui;
+import fr.zcraft.zlib.components.gui.Gui;
+import fr.zcraft.zlib.tools.Callback;
+
+/* Built from recommendations in PR https://github.com/zDevelopers/zLib/pull/14 */
+public class ArrayPromptGui<T> extends ExplorerGui<T> {
+
+	private Callback<T> cb;
+	private String purpose;
+	private T[] datas;
+	private ItemRenderer<T> renderer;
+	private boolean closeOnChoice;
+
+	/**
+	 * @param callback
+	 *            Callback called when the player made a choice
+	 * @param player
+	 *            The player making the choice
+	 * @param purpose
+	 *            The purpose (gui's title)
+	 * @param datas
+	 *            An array of datas to display
+	 * @param renderer
+	 *            Interface for building gui's ItemStack from array values
+	 * @param closeOnChoice
+	 *            Close the interface when the player has choosen if true
+	 */
+	public <A> ArrayPromptGui(Callback<T> callback, Player player, String purpose, T[] datas, ItemRenderer<T> renderer,
```

Comment:
Another possibilty to change this: use a builder pattern, [like this](https://github.com/AmauryCarrade/BallsOfSteel/blob/d351d3af6eb5bf430228349f958ce0c4b88cc999/src/main/java/eu/carrade/amaury/BallsOfSteel/commands/SpheresCommand.java#L105-L108), to configure every non-required option (i.e. everything except player), applying sensible defaults to unset options. The display would be done using a final method in the chain, like in the given example. The constructor would be arguments-less (like 0).